sys/net/application_layer/nanocoap/sock.c

```diff
@@ -128,13 +128,13 @@ int nanocoap_server(sock_udp_ep_t *local, uint8_t *buf, size_t bufsize)
     }
 
     ssize_t res = sock_udp_create(&sock, local, NULL, 0);
-    if (res == -1) {
+    if (res != 0) {
         return -1;
     }
 
     while (1) {
         res = sock_udp_recv(&sock, buf, bufsize, -1, &remote);
-        if (res == -1) {
+        if (res < 0) {
```

Comment:
while you are at it: the else below could be changed into a `else if (res > 0) {` bc if `res == 0` everything is good but nothing was received so no need to call `coap_parse`.

Reply:
I initially didn't want to change the handling of the zero return value. I just changed this nonetheless.

Reply:
thx